gapis/api/vulkan/state_rebuilder.go

```diff
@@ -2441,6 +2441,31 @@ func (sb *stateBuilder) createRenderPass2(rp RenderPassObjectʳ) {
 	subpassDescriptions := []VkSubpassDescription2{}
 	for _, k := range rp.SubpassDescriptions().Keys() {
 		sd := rp.SubpassDescriptions().Get(k)
+		sdPnext := NewVoidᶜᵖ(memory.Nullptr)
+
+		if !sd.DepthStencilResolve().IsNil() {
+			depthStencilResolve := sd.DepthStencilResolve().Get()
+
+			dsrAttachmentRef := NewVkAttachmentReference2ᶜᵖ(sb.MustAllocReadData(
+				NewVkAttachmentReference2(
+					VkStructureType_VK_STRUCTURE_TYPE_ATTACHMENT_REFERENCE_2,
+					NewVoidᶜᵖ(memory.Nullptr),
```

Comment:
AttachmentReference2 here can originally have pNext that we are missing currently.

Reply:
Done, please check this.  Test appears to confirm it is correct.